### ISSUE TITLE
App crashing: Method 'addPostFrameCallback' can not be called on 'SchedulerBinding?'

### ISSUE THREAD

**Issue by iyar-avital** [syncfusion/flutter-widgets#745]
![image](https://user-images.githubusercontent.com/57311417/169214235-4507a72b-3113-4b58-b3e7-86a110710ad1.png)

../../Documents/flutter/.pub-cache/hosted/pub.dartlang.org/syncfusion_flutter_charts-20.1.56/lib/src/chart/base/chart_base.dart:1820:35: Error: Method
'addPostFrameCallback' cannot be called on 'SchedulerBinding?' because it is potentially null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter/packages/flutter/lib/src/scheduler/binding.dart').
Try calling using ?. instead.
        SchedulerBinding.instance.addPostFrameCallback((_) {
                                  ^^^^^^^^^^^^^^^^^^^^
../../Documents/flutter/.pub-cache/hosted/pub.dartlang.org/syncfusion_flutter_charts-20.1.56/lib/src/chart/base/chart_base.dart:1829:37: Error: Method
'addPostFrameCallback' cannot be called on 'SchedulerBinding?' because it is potentially null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../Documents/flutter/packages/flutter/lib/src/scheduler/binding.dart').
Try calling using ?. instead.
          SchedulerBinding.instance.addPostFrameCallback((_) => _refresh());

I am using flutter 2.8.1 and can not upgrade anytime soon due to company policy.
I used version 20.1.55 for one week and everything worked. Today the error appeared.
There is a solution?

**Comment by rivkazi** [comment]
Same problem for me

**Comment by FisherWL** [comment]
why not roll back to 20.1.55?

**Comment by iyar-avital** [comment]
> why not roll back to 20.1.55?

As I wrote, I use 20.55.1.
What do tyu mean?

**Comment by iyar-avital** [comment]
> > why not roll back to 20.1.55?
> 
> As I wrote, I use 20.55.1. What do you mean?

**Comment by FisherWL** [comment]
> > why not roll back to 20.1.55?
> 
> As I wrote, I use 20.55.1. What do you mean?

the 1st line of error message indicates it's caused by 20.1.56,
so I meant by rolling back to 20.1.55 shall solve the problem.

**Comment by iyar-avital** [comment]
> > > why not roll back to 20.1.55?
> > 
> > 
> > As I wrote, I use 20.55.1. What do you mean?
> 
> the 1st line of error message indicates it's caused by 20.1.56, so I meant by rolling back to 20.1.55 shall solve the problem.

Yes, that's what I don't understand. I **have not** upgraded the version to 20.1.56.
What am I missing?

**Comment by FisherWL** [comment]
@iyar-avital 
my guess is you used caret sign (^) in your .yaml file,
changing `syncfusion_flutter_charts: ^20.1.55` to `syncfusion_flutter_charts: 20.1.55` shall solve the problem.

**Comment by iyar-avital** [comment]
> @iyar-avital my guess is you used caret sign (^) in your .yaml file, changing `syncfusion_flutter_charts: ^20.1.55` to `syncfusion_flutter_charts: 20.1.55` shall solve the problem.

![image](https://user-images.githubusercontent.com/57311417/169246398-06b6e9bc-7a2c-4f16-8b68-a9d4f96b3128.png)

![image](https://user-images.githubusercontent.com/57311417/169246467-03d3b3be-8cf9-4a69-b87f-63236ea900de.png)

I have already changed that...
Another idea?

**Comment by FisherWL** [comment]
@iyar-avital 
then maybe try the solution in [https://github.com/syncfusion/flutter-widgets/issues/741](url)?

**Comment by iyar-avital** [comment]
> @iyar-avital if removing ^ along with `flutter clean` and a thorough rebuild then maybe try the solution in #741 ？

I'll try it
thanks for your help

**Comment by tusharlock10** [comment]
Check your flutte version

**Comment by iyar-avital** [comment]
> 19.4.56

it works. thanks.

**Comment by dharanidharandharmasivam** [comment]
Hi @iyar-avital ,

Our chart package with v20.1.56 is compatible with the Flutter latest version (v3.0.0). So, you need to update your Flutter SDK to the latest version if you opt to use the chart package with v20.1.56. Else you can use the lowest version of our package using by having the below code in your pubspec file(don't use the **^** symbol as it will fetch the latest version from the pub).

`syncfusion_flutter_charts: 20.1.55`

Thanks,
Dharani

**Comment by iyar-avital** [comment]
> Hi @iyar-avital ,
> 
> Our chart package with v20.1.56 is compatible with the Flutter latest version. So, you need to update your Flutter SDK to the latest version if you opt to use the chart package with v20.1.56. Else you can use the lowest version of our package using by having the below code in your pubspec file(don't use the **^** symbol as it will fetch the latest version from the pub).
> 
> `syncfusion_flutter_charts: 20.1.55`
> 
> Thanks, Dharani

This error happens also in 20.1.55, we tried it.
can you check it, please?

**Comment by agoldm** [comment]
@dharanidharandharmasivam
It didn't work for me either

**Comment by Marismathan** [comment]
Hi @agoldm ,

You are now using our charts package with v20.1.55. But it refers to chart dependent core(syncfusion_flutter_core) package to the latest version. which is compatible with the Flutter latest version. So we kindly suggest you use the lowest version of our core package too by having the below code in your pubspec file.

`syncfusion_flutter_charts: 20.1.55
syncfusion_flutter_core: 20.1.55`

Thanks,
Marismathan G

**Comment by agoldm** [comment]
@Marismathan It works! Thanks a lot!

**Comment by amirVirtuenetz** [comment]
> > > why not roll back to 20.1.55?
> > 
> > 
> > As I wrote, I use 20.55.1. What do you mean?
> 
> the 1st line of error message indicates it's caused by 20.1.56, so I meant by rolling back to 20.1.55 shall solve the problem.

are you sure if i downgrade the version of syncfusion chart fluttert, it solve my problem?
